src/test/mocks.ts

```diff
@@ -92,7 +111,132 @@ export class MockClient extends MockWatcher implements ErdstallClient {
 		throw new Error("cannot query accounts on mock clients");
 	}
 
+	setMetadata(token: Address, id: bigint, metadata: NFTMetadata): void {
+		this.metadata.set(`${token}:${id}`, metadata);
+	}
+
+	async getNftMetadata(token: Address, id: bigint): Promise<NFTMetadata> {
+		return this.metadata.get(`${token}:${id}`)!;
```

Comment:
Maybe we should return a rejecting promise if the value is undefined? This is how axios would behave, right?